test/functional/test_framework/key.py

```diff
@@ -322,6 +353,40 @@ def verify_ecdsa(self, sig, msg, low_s=True):
             return False
         return True
 
+    def verify_schnorr(self, sig, msg):
+        assert(len(msg) == 32)
+        assert(len(sig) == 64)
+        assert(self.valid)
+        assert(self.compressed)
+        r = int.from_bytes(sig[0:32], 'big')
+        if r >= SECP256K1_FIELD_SIZE:
+            return False
+        s = int.from_bytes(sig[32:64], 'big')
+        if s >= SECP256K1_ORDER:
+            return False
+        e = int.from_bytes(TaggedHash("BIPSchnorr", sig[0:32] + self.get_bytes()[1:33] + msg), 'big') % SECP256K1_ORDER
+        if self.is_positive:
```

Comment:
This seems weird to me. How can we verify a signature for a point which does not have a square Y? Such public keys are not defined in bip-schnorr.

I think it'd be clearer to assert that the pubkey has a square Y before verifying. I'd also suggest adding a `get_xonly_pubkey()` method to `ECKey` so we can directly get the valid bip-schnorr pubkey.

Reply:
> This seems weird to me. How can we verify a signature for a point which does not have a square Y? Such public keys are not defined in bip-schnorr.

Hm in fact it's different in yet another way. 
 - The very pedantic version: bip-schnorr defines public keys to be byte arrays of length 32. "square y" does not make sense for byte arrays. 
 - The less pedantic version: Even if you interpret the byte array as an integer, it's supposed to encode an x-coordinate only. "square y" does not make sense for an x-coordinate.

See also https://github.com/sipa/bips/blob/bip-schnorr/bip-schnorr/reference.py#L99

If we want to be compatible with the existing code here, I'm not sure what the best approach is. Maybe it makes more sense to accept a byte string, then parse it into a point and then use the existing code?

Reply:
`verify_schnorr` has been updated to be a function that takes a pubkey (32-byte array) instead of being a method on `ECPubKey`.

Reply:
I've completely rewritten the Python Schnorr code now, not using ECKey/ECPubKey at all.